aspnetcore/fundamentals/logging.md

```diff
@@ -1,215 +1,446 @@
+* [EventLog](#eventlog)
+* [TraceSource](#tracesource)
+* [Azure App Service](#appservice)
+
+<a id="console"></a>
+### The console provider
+
+The [Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console) provider package sends log output to the console. 
+
+```csharp
+loggerFactory.AddConsole()
+```
+
+[AddConsole overloads](https://docs.microsoft.com/aspnet/core/api/microsoft.extensions.logging.consoleloggerextensions) let you pass in an a minimum log level, a filter function, and a boolean that indicates whether scopes are supported.  Another option is to pass in an `IConfiguration` object, which can specify scopes support and logging levels. 
+
+If you are considering the console provider for use in production, be aware that it has a significant impact on performance.
+
+When you create a new project in Visual Studio, the `AddConsole` method looks like this:
+
+```csharp
+loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+```
+
+This code refers to the `Logging` section of the *appSettings.json* file:
+
+[!code-json[](logging/sample/src/TodoApi/appsettings.json)]
+
+The settings shown limit framework logs to warnings while allowing the app to log at debug level, as explained in the [Log filtering](#log-filtering) section. For more information, see [Configuration](configuration.md).
+
+<a id="debug"></a>
+### The Debug provider
+
+The [Microsoft.Extensions.Logging.Debug](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Debug) provider package writes log output by using the [System.Diagnostics.Debug](https://docs.microsoft.com/dotnet/core/api/system.diagnostics.debug#System_Diagnostics_Debug) class.
+
+```csharp
+loggerFactory.AddDebug()
+```
+
+[AddDebug overloads](https://docs.microsoft.com/aspnet/core/api/microsoft.extensions.logging.debugloggerfactoryextensions) let you pass in a minimum log level or a filter function.
+
+<a id="eventsource"></a>
+### The EventSource provider
 
-You can create your own custom providers as well, to support other logging frameworks or your own internal logging requirements.
+The [Microsoft.Extensions.Logging.EventSource](https://www.nuget.org/packages/Microsoft.Extensions.Logging.EventSource) provider package does event tracing. On Windows it uses [ETW](https://msdn.microsoft.com/library/windows/desktop/bb968803). The provider is cross-platform but there are no event collection and display tools yet for Linux or macOS. 
 
-## Logging Recommendations
+```csharp
+loggerFactory.AddEventSourceLogger()
+```
 
-The following are some recommendations you may find helpful when implementing logging in your ASP.NET Core applications.
+The best way to collect and view logs is to use the [PerfView utility](https://www.microsoft.com/en-us/download/details.aspx?id=28567). There are other tools for viewing ETW logs, but they don't work well with the dynamic provider registration that the ASP.NET Core EventSource provider uses. 
```

Comment:
It's actually not so much that they don't work, but that they have difficulty interpreting them. I'd rephrase as "There are other tools for viewing ETW logs, but PerfView provides the best experience for working with the ETW events emitted by ASP.NET."